src/platform/packages/shared/kbn-unified-tabs/src/components/tab_preview/tab_preview.tsx

```diff
@@ -0,0 +1,202 @@
+      const wouldExtendBeyondRight = rect.left + PREVIEW_WIDTH > windowWidth;
+
+      // Calculate left position based on screen edge constraints
+      let leftPosition = rect.left + window.scrollX;
+
+      if (wouldExtendBeyondRight) {
+        // Align right edge of preview with right edge of window
+        leftPosition = windowWidth - PREVIEW_WIDTH + window.scrollX;
+      }
+
+      setTabPosition({
+        top: rect.bottom + window.scrollY,
+        left: leftPosition,
+      });
+    }
+  }, [showPreview]);
+
+  const onKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === keys.ESCAPE) {
+        setShowPreview(false);
+      }
+    },
+    [setShowPreview]
+  );
+
+  useEvent('keydown', onKeyDown);
+
+  useEffect(() => {
+    return () => {
+      if (previewTimer) {
+        clearTimeout(previewTimer);
+      }
+    };
+  }, [previewTimer]);
+
+  const handleMouseEnter = useCallback(() => {
+    if (stopPreviewOnHover) return;
+
+    const timer = setTimeout(() => {
+      setShowPreview(true);
+    }, previewDelay);
+
+    setPreviewTimer(timer);
+  }, [previewDelay, setShowPreview, stopPreviewOnHover]);
+
+  const handleMouseLeave = useCallback(() => {
+    if (previewTimer) {
+      clearTimeout(previewTimer);
+      setPreviewTimer(null);
+    }
+
+    setShowPreview(false);
+  }, [previewTimer, setShowPreview]);
+
+  return (
+    <div>
+      <span onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave} ref={tabRef}>
+        {children}
+      </span>
+      <EuiPortal>
+        <EuiSplitPanel.Outer
+          grow
+          css={getPreviewContainerCss(euiTheme, showPreview, tabPosition)}
+          data-test-subj={`unifiedTabs_tabPreview_${tabItem.id}`}
+        >
+          <EuiSplitPanel.Inner paddingSize="none" css={getSplitPanelCss(euiTheme)}>
+            <EuiCodeBlock
+              language={getQueryLanguage(tabPreviewData)}
+              transparentBackground
+              paddingSize="none"
+            >
+              {getPreviewQuery(tabPreviewData)}
+            </EuiCodeBlock>
+          </EuiSplitPanel.Inner>
+          <EuiSplitPanel.Inner
+            grow={false}
+            color="subdued"
+            paddingSize="none"
+            css={getSplitPanelCss(euiTheme)}
+          >
+            {tabPreviewData.status === TabStatus.RUNNING ? (
+              <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false}>
+                <EuiFlexItem grow={false}>
+                  <EuiLoadingSpinner />
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiText>{tabItem.label}</EuiText>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            ) : (
+              <EuiHealth color={tabPreviewData.status} textSize="m">
+                {tabItem.label}
+              </EuiHealth>
+            )}
+          </EuiSplitPanel.Inner>
+        </EuiSplitPanel.Outer>
+      </EuiPortal>
```

Comment:
I think it's worth exploring an idea of rendering this popover content layout and calling `getPreviewData` only when a tab is actually hovered. We could do that as a follow up.

Reply:
Great idea!